dask/dataframe/io/parquet.py

```diff
@@ -468,8 +469,10 @@ def _write_fastparquet(df, fs, fs_token, path, write_index=None, append=False,
                  for i in range(df.npartitions)]
 
     write = delayed(_write_partition_fastparquet, pure=False)
-    writes = [write(part, fs, path, filename, fmd, compression, partition_on)
-              for filename, part in zip(filenames, df.to_delayed())]
+    token = str(uuid.uuid1().hex)
+    writes = [write(part, fs, path, filename, fmd, compression, partition_on,
+                    dask_key_name=('write-parquet-%s' % token, i))
```

Comment:
...as easy as that! Perhaps this kind of thing should be documented in the devs pages.